proof/crefine/ARM/Finalise_C.thy

```diff
@@ -1787,12 +1771,6 @@ lemma Arch_finaliseCap_ccorres:
    apply (frule cap_get_tag_isCap_unfolded_H_cap, simp)
   done
 
-lemma ccte_relation_ccap_relation:
-  "ccte_relation cte cte' \<Longrightarrow> ccap_relation (cteCap cte) (cte_C.cap_C cte')"
-  by (clarsimp simp: ccte_relation_def ccap_relation_def
-                     cte_to_H_def map_option_Some_eq2
-                     c_valid_cte_def)
```

Comment:
are these getting removed because they're already proved elsewhere? (referring also to `empty_fail_getNotification`

Reply:
Yeah, exactly. Sorry for mixing them in with unrelated changes, I was just cleaning them up when I noticed them and they were adjacent to lemmas I was already touching.

What are our thoughts for bits of cleanup like this? Optimally they would go in their own commits and even PRs but with how long it takes to rebuild and make changes I feel like that would lead to it almost never happening.

Reply:
Definitely fine to have alongside other commits in one PR.
It's definitely possible to extract these into a separate commit without having to rebuild. One way I'd suggest is to create a branch identical to the current one. Then you can diff against that branch and be sure whatever git-fu you're doing doesn't affect anything.
Via `git add -p`, can add the removals separately, and then commit the rest of the change as a the same commit as before.

The other approach is to notice the lemmas you removed, remove those first as one commit, and then rebase your PR on top of that (and check with backup branch, should be empty diff).

I'm not sure how much it's worth it. If you've only reviewed a few lemmas, it's not going to be very hard, and might be good git practice. We've been ok with leaving small cleanups like this in the past, so your call.